Fix mysql strict mode

MelchnerRoman opened the pull request:
This PR enhances the overall robustness of the codebase by:

Implementing strict mode safety and fallback logic for player data and configuration values, reducing the risk of nil errors and improving reliability.

All changes have been tested to ensure compatibility and stability but my understanding of the whole code base is limited so please review, test and provide feedback as needed.

rmelchner committed:
Add fallbacks for strict mode compatibility

Added fallback values to ensure variables are always defined, preventing runtime errors under Lua's strict mode. This is important because strict mode raises errors when accessing undefined globals, so providing fallbacks improves code robustness and reliability.

rmelchner committed:
update config defaults, add init function fallbacks

Updated player initialization and save logic to use layered fallbacks (session, DB, config defaults) for all key fields. This ensures no nil values are written, improves strict mode compatibility, and enhances reliability on player disconnect and account creation.

iiRedDev commented on config_server.lua:
```diff
@@ -59,8 +59,28 @@ Config.JobInitGrade         = 1
+Config.InitXP               = 10
+
+-- New Default Values for strict mode
+Config.InitInventory        = '[{"amount":1,"meta":[],"name":"WEAPON_MELEE_TORCH"},{"amount":1,"meta":[],"name":"water"},{"amount":1,"meta":[],"name":"bread"}]'
```
For non-developers, they'll have trouble with this line! It would be better to be able to save the items we want at the spawn of natives. ;)

MelchnerRoman committed:
chore: update InitInventory for better readability

MelchnerRoman committed:
fix: bugfixes in Init Player

Remove buggy Coords handling for Guarma and fix default CharID

- Removed usage of player coordinates in config and DB updates, as this caused issues in Guarma.
- Updated incorrect default character ID to use the correct config value.

MelchnerRoman committed:
fix: add back Default Cords

MelchnerRoman committed:
Hardened XP/level logic and fixed notification bug

Hardened XP/level logic and fixed notification bug in player commands

- Updated cl_hud.lua:
  - Made LevelPlayer and XpReturn functions robust against nil, missing, or malformed player/session data and levels.json.
  - Ensured all XP and level calculations use safe defaults and print debug messages if data is missing.
- Updated cl_player_actions.lua:
  - Fixed /mylvl command to always display a valid level or 'N/A' by converting PlayerLevel to a string before concatenation.
  - Prevented runtime errors when player data is missing or incomplete.

iiRedDev committed:
Merge branch 'main' into fix-mysql-strict-mode
